### PR TITLE
lug: add lean4 pkgs

### PR DESCRIPTION
**Pull request by alissa-tung**
I am not sure whether I can use second level paths. If that can work as a namespace I think this would be better...

### CODE REVIEW

File: config.siyuan.yaml [hunk 1]
```diff
@@ -979,3 +979,68 @@ repos:
...
     <<: *oneshot_common
+  # mathlib
+  - type: shell_script
+    script: /worker-script/git.sh
+    interval: 3600
+    name: git/lean4-packages/std4
+    source: https://github.com/leanprover/std4.git
+    path: /srv/data32T/git/lean4-packages/std4
+    serve_mode: git
+    target: https://git.sjtu.edu.cn/sjtug/lean4-packages/std4.git
```
Comment (PhotonQuantum) [inline config.siyuan.yaml:990]: `git.sjtu.edu.cn` is another server hosted by NIC. We don't need it anymore, so you can just remove this line.

File: config.siyuan.yaml [hunk 2]
```diff
@@ -979,3 +979,68 @@ repos:
     target: https://repo.huaweicloud.com/ubuntu-cdimage/
     disabled: true
     <<: *oneshot_common
+  # mathlib
+  - type: shell_script
+    script: /worker-script/git.sh
+    interval: 3600
+    name: git/lean4-packages/std4
+    source: https://github.com/leanprover/std4.git
+    path: /srv/data32T/git/lean4-packages/std4
```
Comment (PhotonQuantum) [inline config.siyuan.yaml:988]: Path must end with `.git` because it's a bare repo.

**Review by PhotonQuantum — CHANGES_REQUESTED** [review]
Don't forget to run `make caddy-gen` after modifications.

**Review by PhotonQuantum — APPROVED** [review]
Looks good! Please rebase it onto the main branch.

**Comment by alissa-tung** [comment]
> Looks good! Please rebase it onto the main branch.

fixed. thank u

**Comment by PhotonQuantum** [comment]
Would you be willing to provide a guide on how to use this mirror? https://github.com/sjtug/portal

**Comment by alissa-tung** [comment]
> Would you be willing to provide a guide on how to use this mirror? https://github.com/sjtug/portal

Yes. I am writing one

**Pull request merged**